tests/RetryMiddlewareTest.php

```diff
@@ -81,6 +83,33 @@ public function testDeciderIgnoresPHPError()
         }
     }
 
+    public function testDeciderRetriesWhenCurlErrorCodeMatches()
+    {
+        $decider = RetryMiddleware::createDefaultDecider();
+        $command = new Command('foo');
+        $request = new Request('GET', 'http://www.example.com');
+        if ((string) ClientInterface::VERSION[0] === '6') {
+            $previous = new ConnectException(
+                'test',
+                $request,
+                null,
+                ['errno' => CURLE_RECV_ERROR]
+            );
+        } elseif ((string) ClientInterface::VERSION[0] === '5') {
+            $previous = new ConnectException(
+                'cURL error ' . CURLE_RECV_ERROR . ': test',
+                new \GuzzleHttp\Message\Request('GET', 'http://www.example.com')
+            );
+        }
+        $err = new AwsException(
+            'e',
+            $command,
+            ['connection_error' => false],
+            $previous
+        );
+        $this->assertTrue($decider(0, $command, $request, null, $err));
```

Comment:
My impression from https://github.com/aws/aws-sdk-php/issues/924#issuecomment-355646600 was that `CURLE_RECV_ERROR` would not be retried by default. Did you mean it wouldn't be retried by Guzzle but would be retried by the SDK?

Reply:
I've left a [new comment](https://github.com/aws/aws-sdk-php/issues/924#issuecomment-356445082) on that thread (reproduced below) that states the change:

> After some further discussion based around the operations being handled, it is believed we can retry this at the SDK level since it's not safe to do at the Guzzle level. Support is being added for automatic retry of `CURLE_RECV_ERROR` via #1463.